build: bootstrap doesn't use vendored modules

**Describe the problem**

The build process seems to always download the bootstrap go modules from the internet, despite them being vendored.

**To Reproduce**

Using an official stable release tarball, try to build without network, eg. in a netns:
```
 /tmp/cockroach-v20.2.7 $ unshare -fcn
 /tmp/cockroach-v20.2.7 $ make build
make -C src/github.com/cockroachdb/cockroach build
make[1]: Entering directory '/tmp/cockroach-v20.2.7/src/github.com/cockroachdb/cockroach'
GOPATH set to /tmp/cockroach-v20.2.7
which: no ccache in (/home/implr/google-cloud-sdk/bin:/opt/chefdk/bin:/home/implr/go/bin:/home/implr/bin/:/home/implr/.local/bin/:/home/implr/.cabal/bin/:/home/implr/.cargo/bin/:/home/implr/.nix-profile/bin:/home/implr/.nix-profile/bin:/nix/var/nix/profiles/default/bin:/opt/stuffit/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/bin:/usr/lib/llvm/11/bin)
Warning: 'ccache' not found, consider installing it for faster builds
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
Detected change in build system. Rebooting make.
Running make with -j8
GOPATH set to /tmp/cockroach-v20.2.7
which: no ccache in (/home/implr/google-cloud-sdk/bin:/opt/chefdk/bin:/home/implr/go/bin:/home/implr/bin/:/home/implr/.local/bin/:/home/implr/.cabal/bin/:/home/implr/.cargo/bin/:/home/implr/.nix-profile/bin:/home/implr/.nix-profile/bin:/nix/var/nix/profiles/default/bin:/opt/stuffit/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/bin:/usr/lib/llvm/11/bin)
Warning: 'ccache' not found, consider installing it for faster builds
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
gitdir=$(git rev-parse --git-dir 2>/dev/null || true); \
if test -n "$gitdir"; then \
   git submodule update --init --recursive; \
fi
mkdir -p bin
mkdir -p pkg/sql/parser/gen
touch bin/.submodules-initialized
set -euo pipefail; \
awk '/func.*sqlSymUnion/ {print $(NF - 1)}' pkg/sql/parser/sql.y | \
sed -e 's/[]\/$*.^|[]/\\&/g' | \
sed -e "s/^/s_(type|token) <(/" | \
awk '{print $0")>_\\1 <union> /* <\\2> */_"}' > pkg/sql/parser/gen/types_regex.tmp; \
sed -E -f pkg/sql/parser/gen/types_regex.tmp < pkg/sql/parser/sql.y | \
awk -f pkg/sql/parser/replace_help_rules.awk | \
sed -Ee 's,//.*$,,g;s,/[*]([^*]|[*][^/])*[*]/, ,g;s/ +$//g' > pkg/sql/parser/gen/sql-gen.y.tmp || rm pkg/sql/parser/gen/sql-gen.y.tmp
cd /tmp/cockroach-v20.2.7/src/github.com/cockroachdb/cockroach/c-deps/jemalloc && autoconf
cd /tmp/cockroach-v20.2.7/src/github.com/cockroachdb/cockroach/c-deps/libedit && autoconf
rm -rf /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/protobuf
rm -rf /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/snappy
go install -v ./pkg/cmd/prereqs
mkdir -p /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/protobuf
mkdir -p /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/snappy
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
cd /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/protobuf && cmake -DCMAKE_TARGET_MESSAGES=OFF   -Dprotobuf_BUILD_TESTS=OFF /tmp/cockroach-v20.2.7/src/github.com/cockroachdb/cockroach/c-deps/protobuf/cmake \
  -DCMAKE_BUILD_TYPE=Release
cd /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/snappy && cmake -DCMAKE_TARGET_MESSAGES=OFF   /tmp/cockroach-v20.2.7/src/github.com/cockroachdb/cockroach/c-deps/snappy \
  -DCMAKE_BUILD_TYPE=Release
rm -rf /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/proj
mkdir -p /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/proj
cd /tmp/cockroach-v20.2.7/native/x86_64-pc-linux-gnu/proj && cmake  -DCMAKE_TARGET_MESSAGES=OFF   /tmp/cockroach-v20.2.7/src/github.com/cockroachdb/cockroach/c-deps/proj -DCMAKE_BUILD_TYPE=Release -DBUILD_LIBPROJ_SHARED=OFF
go: cloud.google.com/go@v0.34.0: Get "https://proxy.golang.org/cloud.google.com/go/@v/v0.34.0.mod": dial tcp: lookup proxy.golang.org: Temporary failure in name resolution
CMake Deprecation Warning at CMakeLists.txt:12 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


make[1]: *** [Makefile:371: bin/.bootstrap] Error 1
make[1]: *** Waiting for unfinished jobs....
go: cloud.google.com/go@v0.34.0: Get "https://proxy.golang.org/cloud.google.com/go/@v/v0.34.0.mod": dial tcp: lookup proxy.golang.org: Temporary failure in name resolution
make[1]: *** [Makefile:1795: bin/prereqs] Error 1

```

**Expected behavior**
Build succeeds without network.


**Environment:**
 - CockroachDB version: https://binaries.cockroachdb.com/cockroach-v20.2.7.src.tgz
 - Distro: gentoo
 - go version go1.16.2 linux/amd64

--- a/Makefile
+++ b/Makefile
@@ -253,7 +253,7 @@ $(info GOPATH set to $(GOPATH))
 
 # We install our vendored tools to a directory within this repository to avoid
 # overwriting any user-installed binaries of the same name in the default GOBIN.
-GO_INSTALL := GOBIN='$(abspath bin)' GOFLAGS= $(GO) install
+GO_INSTALL := GOBIN='$(abspath bin)' GOFLAGS= $(GO) install $(GOMODVENDORFLAGS)
 
 # Prefer tools we've installed with go install and Yarn to those elsewhere on
 # the PATH.
@@ -1638,7 +1638,7 @@ pkg/sql/opt/rule_name.og.go: $(optgen-defs) $(optgen-norm-rules) $(optgen-xform-
 	optgen -out $@ rulenames $(optgen-defs) $(optgen-norm-rules) $(optgen-xform-rules)
 
 pkg/sql/opt/rule_name_string.go: pkg/sql/opt/rule_name.go pkg/sql/opt/rule_name.og.go bin/.bootstrap
-	stringer -output=$@ -type=RuleName $(filter %.go,$^)
+	GOFLAGS=$(GOMODVENDORFLAGS) stringer -output=$@ -type=RuleName $(filter %.go,$^)
 
 pkg/sql/opt/xform/explorer.og.go: $(optgen-defs) $(optgen-xform-rules) bin/optgen
 	optgen -out $@ explorer $(optgen-defs) $(optgen-xform-rules)

